common/components/test/utils.test.js

```diff
@@ -167,5 +168,30 @@ describe("utils", () => {
     let test = array.join(testArray, ",");
     let testShouldEqual: Array<string> = ["test1", ",", "test2"];
     expect(test).toEqual(testShouldEqual);
-  });
+  });  
+
+  test("utils.navigateToTopOfPage", () => {
+    global.scrollTo = jest.fn();
+    utils.navigateToTopOfPage();
+    expect(global.scrollTo).toBeCalledWith(0, 0);
+  }); 
+
+  test("utils.unescapeHtml", () => {
+    const rawStringHtmlContent = "&lt;p&gt;Function unescapeHtml&#x27;s test on &lt;a href=&#x27;fake url&#x27;&gt;something&lt;/a&gt;something&lt;/p&gt;";
+    const tranformedStringHtmlContent = utils.unescapeHtml(rawStringHtmlContent);
+    expect(tranformedStringHtmlContent).toEqual("<p>Function unescapeHtml's test on <a href='fake url'>something</a>something</p>");
+  }); 
+
+  test("utils.pluralize", () => {
+    let word = utils.pluralize("apple", "apples", 1);
+    expect(word).toEqual("apple");
+    word = utils.pluralize("apple", "apples", '1');
+    expect(word).toEqual("apples");
+    word = utils.pluralize("apple", "apples", [1]);
+    expect(word).toEqual("apples");
```

Comment:
While it can be good to test functions with unexpected input, with unit tests we should stick with defined behavior.  The 'count' variable is expected to be a number, and if other types are used we don't have defined behavior for them.  Testing against strings or arrays are essentially testing our implementation details, not the contract of the function itself, and so I would remove these tests.

Reply:
Thank you for your suggestions. Removed :).